src/features/sign-up/ui/SignUpForm.tsx

```diff
@@ -1,28 +1,53 @@
+/* eslint-disable react-native/no-inline-styles */
 import { FC } from 'react';
 
 import { FormProvider } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
 
+import { IdentityModel } from '@app/entities/identity';
 import { useAppForm } from '@app/shared/lib';
-import { Button, Text, Box, BoxProps, YStack, XStack } from '@shared/ui';
-import { InputField, CheckBoxField } from '@shared/ui/form';
+import {
+  Text,
+  Box,
+  BoxProps,
+  YStack,
+  XStack,
+  ProgressButton,
+} from '@shared/ui';
+import { InputField, CheckBoxField, ErrorMessage } from '@shared/ui/form';
 
 import { SignUpFormSchema } from '../model';
 
-const SignUpForm: FC<BoxProps> = props => {
+type Props = BoxProps & {
+  onLoginSuccess: () => void;
```

Comment:
I think `onSignUpSuccess` name might be more suitable for that component.
or `onRegistrationSuccess`